conans/client/remote_manager.py

```diff
@@ -161,10 +204,9 @@ def addfile(name, file_info, tar):
 def uncompress_files(files, folder, name):
     try:
         for file_name, content in files:
-            if os.path.basename(file_name) != name:
-                save(os.path.join(folder, file_name), content)
-            else:
-                #  Unzip the file
+            save(os.path.join(folder, file_name), content)
```

Comment:
There is one important thing I do not like with this approach. For saving a few seconds in the check before re-upload, we are storing a potentially very large tgz file, which can highly increase storage requirements in the conan local cache.

My local cache tends to bloat quickly when installing packages like Boost or Qt, and I sometimes have to clean it to release space. I know disc storage is cheap, but not that much for SSD, so no totally convinced.

Would it be possible just to store a file with the hash of the tgz file, instead of the full tgz? For checking purposes is the same, and the logic for adding and removing such a file would be the same as the tgz, so everything should keep consistent, as in your proposal.